src/main/java/frc/robot/commands/MoveToGamepiece.java

```diff
@@ -13,48 +13,46 @@ public class MoveToGamepiece extends Command {
     private SwerveDrivetrain drivetrain;
     private Vision vision;
     private double startTime;
-    private double timeSincePieceLoss;
     private final ProfiledPIDController turningPIDController;
     private final ProfiledPIDController movingPIDController;
+    private ChassisSpeeds driveStates;
+    private double ychange;
+
 
     public MoveToGamepiece(SwerveDrivetrain drivetrain, Vision vision) {
         this.drivetrain = drivetrain;
         this.vision = vision;
         addRequirements(drivetrain);
         TrapezoidProfile.Constraints constraints = new TrapezoidProfile.Constraints(Constants.GAMEPIECE_MAX_VELOCITY, Constants.GAMEPIECE_MAX_ACCELERATION);
-        double tP = 0.04;
+        double tP = 0.015;
+        double tD = 0.0015;
         double mP = 0.05;
-        turningPIDController = new ProfiledPIDController(tP, 0, 0, constraints);
+        turningPIDController = new ProfiledPIDController(tP, 0, tD, constraints);
         movingPIDController = new ProfiledPIDController(mP, 0, 0, constraints);
 
     }
 
     @Override
     public void initialize() {
         startTime = Timer.getFPGATimestamp();
-        timeSincePieceLoss = Timer.getFPGATimestamp();
     }
 
     @Override
     public void execute() {
-        ChassisSpeeds driveStates;
-        double ychange = vision.getPieceOffestAngleY() - Constants.LIMELIGHT_TURN_TO_PIECE_DESIRED_Y;
+        ychange = vision.getPieceOffestAngleY() - Constants.LIMELIGHT_TURN_TO_PIECE_DESIRED_Y;
         if (vision.isPieceSeen() && (Math.abs(ychange) > Constants.MOVE_TO_GAMEPIECE_THRESHOLD)) {
-            timeSincePieceLoss = Timer.getFPGATimestamp();
-            driveStates = new ChassisSpeeds(movingPIDController.calculate(ychange), 0, turningPIDController.calculate(vision.getPieceOffestAngleX()));
-        }
-        else driveStates = drivetrain.createChassisSpeeds(-0.6, 0.0, 0.0, false);
+        driveStates = new ChassisSpeeds(movingPIDController.calculate(ychange), 0, turningPIDController.calculate(vision.getPieceOffestAngleX() - Constants.LIMELIGHT_TURN_TO_PIECE_DESIRED_X));
         drivetrain.drive(driveStates);
+        }
     }
 
     @Override
     public boolean isFinished() {
-        return ((Timer.getFPGATimestamp() - timeSincePieceLoss >= Constants.TIMEOUT_AFTER_PIECE_NOT_SEEN) || (Timer.getFPGATimestamp() - startTime > Constants.MOVE_TO_GAMEPIECE_TIMEOUT));
+        return (Timer.getFPGATimestamp() - startTime > Constants.MOVE_TO_GAMEPIECE_TIMEOUT) || (!vision.isPieceSeen() || (Math.abs(ychange) < Constants.MOVE_TO_GAMEPIECE_THRESHOLD));
```

Comment:
Can you add a Timeout counter to this so that we can log when it times out.

Reply:
don't add a timeoutConter here, the goal of the timeoutCounter is to detect a bad sensor. This is not the case here, this command may time out and it is ok.